subt_ign/src/path_tracer.cc

```diff
@@ -0,0 +1,354 @@
+/*
+ * Copyright (C) 2020 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+#include "path_tracer.hh"
+
+//////////////////////////////////////////////////
+Processor::Processor(const std::string &_path, int _stepSleepMs)
+  : stepSleepMs(_stepSleepMs)
+{
+  // Create the transport node with the partition used by simulation
+  // world.
+  ignition::transport::NodeOptions opts;
+  opts.SetPartition("PATH_TRACER");
+  this->markerNode = new ignition::transport::Node(opts);
```

Comment:
Changing this to a smart pointer would allow you to get ride of the destructor.

Reply:
Fixed in 4890364ef47509e855e24a38e3fd5e5b3a6da1c5